x/kavadist/genesis_test.go

```diff
@@ -0,0 +1,66 @@
+package kavadist_test
+
+import (
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/suite"
+
+	tmtime "github.com/tendermint/tendermint/types/time"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/kava-labs/kava/x/kavadist"
+	testutil "github.com/kava-labs/kava/x/kavadist/testutil"
+	"github.com/kava-labs/kava/x/kavadist/types"
+)
+
+type genesisTestSuite struct {
+	testutil.Suite
+}
+
+func (suite *genesisTestSuite) TestInitGenesis_ValidationPanic() {
+	invalidState := types.NewGenesisState(
+		types.Params{
+			Active: true,
+			Periods: []types.Period{
+				{
+					Start:     time.Date(2021, 1, 1, 1, 1, 1, 1, time.UTC),
+					End:       tmtime.Canonical(time.Unix(1, 0)),
+					Inflation: sdk.OneDec(),
+				},
+			},
+		},
+		tmtime.Canonical(time.Unix(1, 0)),
+	)
+
+	suite.Require().Panics(func() {
+		kavadist.InitGenesis(suite.Ctx, suite.Keeper, suite.AccountKeeper, invalidState)
+	}, "expected init genesis to panic with invalid state")
+}
+
+func (suite *genesisTestSuite) TestInitAndExportGenesis() {
+	state := types.NewGenesisState(
+		types.Params{
+			Active: true,
+			Periods: []types.Period{
+				{
+					Start:     time.Date(2021, 1, 1, 1, 1, 1, 1, time.UTC),
+					End:       time.Date(2021, 2, 1, 1, 1, 1, 1, time.UTC),
+					Inflation: sdk.OneDec(),
+				},
+			},
+		},
+		tmtime.Canonical(time.Unix(1, 0)),
+	)
+
+	kavadist.InitGenesis(suite.Ctx, suite.Keeper, suite.AccountKeeper, state)
+	suite.Require().Equal(state.Params, suite.Keeper.GetParams(suite.Ctx))
+
```

Comment:
nit: I'd maybe not use the `DefaultPreviousBlockTime` in `state` and instead check a normal time is imported and exported as expected. During mainnet upgrades, the genesis state `PreviousBlockTime` will never be the default.

For context, using `DefaultPreviousBlockTime` in the genesis state basically means the previous block time is unknown and should start from the next available block time. This is really just a convenience feature for starting chains from scratch.